Amethyst/Managers/WindowManager.swift

```diff
@@ -13,32 +13,31 @@ import RxSwiftExt
 import Silica
 import SwiftyJSON
 
-enum WindowChange {
-    case add(window: SIWindow)
-    case remove(window: SIWindow)
-    case focusChanged(window: SIWindow)
-    case windowSwap(window: SIWindow, otherWindow: SIWindow)
+enum WindowChange<Window: WindowType> {
+    case add(window: AnyWindow<Window>)
+    case remove(window: AnyWindow<Window>)
+    case focusChanged(window: AnyWindow<Window>)
+    case windowSwap(window: AnyWindow<Window>, otherWindow: AnyWindow<Window>)
     case unknown
 }
 
 // These are the possible actions that the mouse might be taking (that we care about).
 //  We use this enum to convey some information about the window that the mouse
 //  might be interacting with.
-enum MouseState {
+enum MouseState<Window: WindowType> {
     case pointing
     case clicking
     case dragging
-    case moving(window: SIWindow)
+    case moving(window: AnyWindow<Window>)
     case resizing(screen: NSScreen, ratio: CGFloat)
     case doneDragging(atTime: Date)
 }
 
 // MouseStateKeeper will need a few things to do its job effectively
 protocol MouseStateKeeperDelegate: class {
-    func focusedScreenManager() -> ScreenManager?
-    func windows(on screen: NSScreen) -> [SIWindow]
-    func switchWindow(_ window: SIWindow, with otherWindow: SIWindow)
-    var lastReflowTime: Date { get }
+    associatedtype Window: WindowType
+    func recommendMainPaneRatio(_ ratio: CGFloat)
+    func swapDraggedWindowWithDropzone(_ draggedWindow: AnyWindow<Window>)
```

Comment:
I assume that this is just delegating the "which window is under the one being dragged" calculation to some other function that we have access to in this scope. Right?